src/main/scala/viper/gobra/frontend/PackageResolver.scala

```diff
@@ -109,7 +95,13 @@ object PackageResolver {
       // to these resources.
       resource <- resourceUri.getScheme match {
         case s if s == fileUriScheme => Some(FileResource(Paths.get(resourceUri).toFile))
-        case s if s == jarUriScheme => Some(BaseJarResource(resourceUri, path))
+        case s if s == jarUriScheme =>
+          val fs = try {
+            FileSystems.newFileSystem(resourceUri, Collections.emptyMap[String, Any]())
+          } catch {
+            case _: FileSystemAlreadyExistsException => FileSystems.getFileSystem(resourceUri)
+          }
+          Some(JarResource(fs, path))
```

Comment:
Instead of relying on the exceptions thrown by the methods in `FileSystems`, I first tried to keep track of open file systems in the `PackageResolver` object with a `Map[URI, FileSystem]`. However, this would still lead to a `FileSystemAlreadyExistsException` when instantiating a FS with the URI of the `src/main/resources/builtin` folder. It would work OK with the URI from `src/main/resources/stubs` though. I assume that the first filesystem is initialized before package resolution.

Reply:
Are both calls to `FileSystems.newFileSystem` with `src/main/resources/builtin` being made by the code listed in this class? I cannot get my head around why the map does not work

Reply:
Yes, in my experiments with Map, this was the only place where I both created a new FileSystem and updated the corresponding entry in the Map. The first instantiation of a new FileSystem with the URI for `src/main/resources/builtin` would crash.

Reply:
Might it be that Java complains about having two filesystems for the same JAR? I mean they point to distinct folders but have a common root (i.e. `resources`). Might be force trying in the future to use the same filesystem for `builtin` and `stubs`.

Reply:
I believe that the URIs for `builtin` and `stubs` are different. As such, I would be surprised if that were the case

Reply:
They are, however maybe Java maps both of them to the same file system and thus throws an exception

Reply:
Yes that might be the case indeed!